third_party/blink/renderer/core/html/media/html_media_element.cc

```diff
@@ -405,26 +405,53 @@ std::ostream& operator<<(std::ostream& stream,
   return stream << static_cast<void const*>(&media_element);
 }
 
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+// Checks for progressive formats served by the YouTube H5 player.
+// These formats have a mime type of "video/mp4", and lists both audio and
+// video codecs under the "codecs" parameter.
+bool IsProgressiveFormat(const ContentType& content_type) {
+  const String type = content_type.GetType();
+  const String codecs = content_type.Parameter("codecs");
+
+  if (type.empty() && codecs.empty()) {
+    return false;
+  }
+
+  Vector<String> split_codecs;
+  const String separator(", ");
```

Comment:
Just to double check if this can handle queries without space after ',', e.g. 'video/mp4; codecs="avc1.42001E,mp4a.40.2"', which are also valid.

Since we are not checking the exact value, splitting on ',' can handle cases with or without extra space.  In either case, we should comment here mentioning that this is not comprehensive.

Also, is this function called by the H5 player with a single codec?

Reply:
Created a new PR https://github.com/youtube/cobalt/pull/4624 that strips whitespace.

Now that progressive is disabled, this function will be called on single codec formats as well. There may be a performance hit that I haven't analyzed.